docs/development/design/github-application.rst

```diff
@@ -0,0 +1,306 @@
+  organization permissions, the organization owner must approve the application.
+
+  Account owners can use a GitHub App in one account without granting access to
+  another ... A GitHub App remains installed if the person who set it up leaves
+  the organization.
+
+  The installation token from a GitHub App loses access to resources if an admin
+  removes repositories from the installation.
+
+  Installation access tokens are limited to specified repositories with the
+  permissions chosen by the creator of the app.
+
+  GitHub Apps can request separate access to issues and pull requests without
+  accessing the actual contents of the repository.
+
+  A GitHub App receives a webhook event when an installation is changed or
+  removed. This tells the app creator when they've received more or less access to
+  an organization's resources.
+
+  A GitHub App can request an installation access token by using a private key
+  with a JSON web token format out-of-band.
+
+  An installation token identifies the app as the GitHub Apps bot, such as
+  @jenkins-bot.
+
+  GitHub Apps can authenticate on behalf of the user, which is called
+  user-to-server requests. The flow to authorize is the same as the OAuth App
+  authorization flow. User-to-server tokens can expire and be renewed with a
+  refresh token
+
+(from https://docs.github.com/en/developers/apps/getting-started-with-apps/differences-between-github-apps-and-oauth-apps)
+
+
+GitHub Application requirements
+-------------------------------
+
+:Repository permissions:
+   - Metadata: Read-only (mandatory)
+   - Contents: Read-only
+   - Commit statuses: Read & Write
+   - Pull requests: Read-only
+
+:Organization permissions:
+   - Members: Read-only
+
+:User permissions:
+   - Email addresses: Read-only
+
+:Events subscribed:
+   - Meta (When this App is deleted and the associated hook is removed)
+   - Create (Branch or tag created)
+   - Delete (Branch or tag deleted)
+   - Push (Git push to a repository)
+   - Pull request (Pull request opened, closed, reopened, ...)
+
+
+Authentication as GitHub Application
+------------------------------------
+
+https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps
+
+Generate the JWT on Python:
+
+.. code:: python
+
+   import datetime
+   import jwt
+   GITHUB_APP_ID = 134302
+
+   # content of PEM file downloaded from GH Application settings
+   private_key = b'''-----BEGIN RSA PRIVATE KEY-----'''
+
+   jwt.encode({"iat": datetime.datetime.utcnow() - datetime.timedelta(seconds=60), "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=5), "iss": GITHUB_APP_ID}, private_key, algorithm="RS256")
+
+Use the token generated in the previous step as authorization in the cURL command:
+
+.. code:: bash
+
+   $ curl -H "Authorization: Bearer <JWT>" -H "Accept: application/vnd.github.v3+json" https://api.github.com/app
+
+
+.. note::
+
+   For some reason this is not working and I'm getting:
```

Comment:
It says the app needs to be installed on at least 1 organization. That could be it.